tests/providers/amazon/aws/operators/test_emr_add_steps.py

```diff
@@ -207,3 +207,22 @@ def test_init_with_nonexistent_cluster_name(self):
             with pytest.raises(AirflowException) as ctx:
                 operator.execute(self.mock_context)
             assert str(ctx.value) == f"No cluster found for name: {cluster_name}"
+
+    def test_wait_for_completion(self):
+        def check_wait_for_completion(**kwargs):
+            return kwargs.get("wait_for_completion")
+
+        wait_for_completion = False
+        with patch(
+            "airflow.providers.amazon.aws.hooks.emr.EmrHook.add_job_flow_steps"
+        ) as mock_add_job_flow_steps:
+            mock_add_job_flow_steps.side_effect = check_wait_for_completion
+            operator = EmrAddStepsOperator(
+                task_id="test_task",
+                job_flow_id="j-8989898989",
+                aws_conn_id="aws_default",
+                dag=DAG("test_dag_id", default_args=self.args),
+                wait_for_completion=wait_for_completion,
+            )
+
+            assert operator.execute(self.mock_context) == wait_for_completion
```

Comment:
I m not sure about your unit test, I hardly understand the logic. Here is how I'd write it:

```suggestion
    @mock.patch.object(EmrHook, "add_job_flow_steps")
    def test_wait_for_completion(self, mock_add_job_flow_steps):
        job_flow_id = "j-8989898989"
        operator = EmrAddStepsOperator(
            task_id="test_task",
            job_flow_id=job_flow_id,
            aws_conn_id="aws_default",
            dag=DAG("test_dag_id", default_args=self.args),
            wait_for_completion=False,
        )

        mock_add_job_flow_steps.assert_called_once_with(
            job_flow_id=job_flow_id,
            steps=[],
            wait_for_completion=False,
        )
```

Reply:
I thought I needed a test code to see if the parameter was well included in the `add_job_flow_steps` function.
Because this problem was found to have occurred  `wait_for_completion` variable entered the fixed value to `add_job_flow_steps`, so i think parameter check was required. If `wait_for_completion` parameter is not working, I thought it is problem of `add_job_flow_steps` function, not the `EmrAddStepsOperator`.

I'm not sure if I'm right either, so I'd like you to tell me if you have any other opinions.

Reply:
In my suggestion you are checking that the value of `wait_for_completion` is passed to the hook. This check is done on line `222` with `mock_add_job_flow_steps.assert_called_once_with`

Reply:
Thank you for your kind explanation. Thanks to you, I learned that the `assert_called_once_with` method is check that it was called with the correct arguments.